include/seqan3/range/view/kmer_hash.hpp

```diff
@@ -12,47 +12,639 @@
+    kmer_hash_view(kmer_hash_view && rhs)                  = default; //!< Defaulted.
+    kmer_hash_view & operator=(kmer_hash_view const & rhs) = default; //!< Defaulted.
+    kmer_hash_view & operator=(kmer_hash_view && rhs)      = default; //!< Defaulted.
+    ~kmer_hash_view()                                      = default; //!< Defaulted.
+    
+    //!\brief Construct from a view and a given shape.
+    kmer_hash_view(urng_t urange_, shape const s_) : urange{std::move(urange_)}, shape_{s_}
+    {}
+        
+    //!\brief Construct from a non-view that can be view-wrapped and a given shape.
+    template <std::ranges::ViewableRange orng_t>
+    kmer_hash_view(orng_t && urange_, shape const s_) : 
+        urange{std::view::all(std::forward<orng_t>(urange_))}, shape_{s_}
+    {}
+    //!\}
+
+    /*!\name Iterators
+     * \{
+     */
+    /*!\brief Returns an iterator to the first element of the range.
+     * \returns Iterator to the first element.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    auto begin() noexcept
+    {
+        return iterator{std::ranges::begin(urange), *this};
+    }
+    
+    //!\copydoc begin()
+    auto begin() const noexcept
+    //!\cond
+        requires ConstIterableRange<urng_t>
+    //!\endcond
+    {
+        return const_iterator{std::ranges::begin(urange), *this};
+    }
+    
+    //!\copydoc begin()
+    auto cbegin() const noexcept
+    //!\cond
+        requires ConstIterableRange<urng_t>
+    //!\endcond
+    {
+        return const_iterator{std::ranges::begin(urange), *this};
+    }
+    
+    /*!\brief Returns an iterator to the element following the last element of the range.
+     * \returns Iterator to the end.
+     *
+     * \details
+     *
+     * This element acts as a placeholder; attempting to dereference it results in undefined behaviour.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    auto end() noexcept
     {
-        return detail::adaptor_from_functor{*this, k};
+        return std::ranges::end(urange);
     }
+    
+    //!\copydoc end()
+    auto end() const noexcept
+    //!\cond
+        requires ConstIterableRange<urng_t>
+    //!\endcond
+    {
+        return std::ranges::end(urange);
+    }
+    
+    //!\copydoc end()
+    auto cend() const noexcept
+    //!\cond
+        requires ConstIterableRange<urng_t>
+    //!\endcond
+    {
+        return std::ranges::end(urange);
+    }
+    //!\}
+    
+    /*!\brief Returns the number of elements in the view.
+     * \returns The number of elements in the view.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     */
+    auto size() const noexcept
+    //!\cond
+        requires std::SizedSentinel<std::ranges::sentinel_t<urng_t>, std::ranges::iterator_t<urng_t>>
+    //!\endcond
+    {
+        return end() - begin();
+    }
```

Comment:
This is provided by the view_interface already

Reply:
For some reason the concept check fails if I don't provide this overload.

Reply:
I meant that you should ensure that the iterator and the sentinel of your view should model SizedSentinel. (not checking that on the underlying range)